### PR TITLE
Support alternate Message implemetation

### PR DESCRIPTION
**Pull request by jhogg**
Adds support to basic.py/channel/get and channel/build_inbound_message to all specification of an alternate Message class, derived from BaseMessage, for the creation of messages read.

I was not sure the best way to implement test case(s) for an alternate Message implementation, so there are no test cases submitted at this time.

### CODE REVIEW

File: amqpstorm/basic.py [hunk 1]
```diff
@@ -76,10 +78,16 @@ def get(self, queue='', no_ack=False, to_dict=False, auto_decode=True):
...
+        if message_impl:
+            if not issubclass(message_impl, BaseMessage):
+                raiseAMQPInvalidArgument('message_impl should be derived from BaseMessage')
+        else:
+            message_impl = Message
         get_frame = specification.Basic.Get(queue=queue,
                                             no_ack=no_ack)
         with self._channel.lock and self._channel.rpc.lock:
-            message = self._get_message(get_frame, auto_decode=auto_decode)
+            message = self._get_message(get_frame, auto_decode=auto_decode,  \
```
Comment (eandersson) [inline amqpstorm/basic.py:89]: You don't need the \ here at the end of this line.

File: amqpstorm/channel.py [hunk 2]
```diff
@@ -135,16 +136,26 @@ def build_inbound_messages(self, break_on_empty=False, to_tuple=False,
...
         :raises AMQPChannelError: Raises if the channel encountered an error.
         :raises AMQPConnectionError: Raises if the connection
                                      encountered an error.
 
         :rtype: :py:class:`generator`
         """
         self.check_for_errors()
+        if message_impl:
+            if not issubclass(message_impl, BaseMessage):
+                raise AMQPInvalidArgument('message_impl must derive " \
```
Comment (eandersson) [inline amqpstorm/channel.py:152]: And here.

File: amqpstorm/basic.py [hunk 3]
```diff
@@ -76,10 +78,17 @@ def get(self, queue='', no_ack=False, to_dict=False, auto_decode=True):
         elif self._channel.consumer_tags:
             raise AMQPChannelError("Cannot call 'get' when channel is "
                                    "set to consume")
+        if message_impl:
+            if not issubclass(message_impl, BaseMessage):
+                raiseAMQPInvalidArgument('message_impl should be derived " \
```
Comment (eandersson) [inline amqpstorm/basic.py:83]: Same here.

**Comment by eandersson** [comment]
This looks fine to me. I'll try to sit down and test it out sometime this week.

**Comment by jhogg** [comment]
Travis and flake8 all happy now.  Sorry for the rapid commits, we're not running Travis and I don't have everything on my local system to handle your tests.   I'll try and address both before any additional patches.

Let me know if you have any questions or would like anything fixed.
Jay

**Comment by eandersson** [comment]
I didn't get the time to look last week, hopefully this week. Went camping 😄.

I like this idea and maybe something we should consider implementing something similar in other areas as well, e.g. allow hooks to collect stats.

**Comment by jhogg** [comment]
No issue.  Enjoy any time you can get these days away from reality.  I have a couple other changes in-flight and I'll give you some background on why I setup Message as an implementation if you wnat - just add @getaroom.com to my github name and hit me up.

**Comment by eandersson** [comment]
Sorry still haven't been able to get to this. At least I pushed 2.8.2 which has your first change in there! Hopefully sometime this week.

**Comment by jhogg** [comment]
@eandersson Any thoughts on pulling in this change?

**Comment by eandersson** [comment]
Sorry @jhogg I'll try to get to it this week.

**Comment by eandersson** [comment]
To get the ball rolling here I created a new branch called 2.x and the main branch will now be working towards the next major release 3.0 and merged this PR.

**Pull request merged**